apps/src/blockly/addons/cdoFieldParameter.ts

```diff
@@ -0,0 +1,265 @@
+          this.setValue(id);
+          break;
+      }
+    }
+  }
+
+  /**
+   * Find the selected parameter, remove it from the procedure model,
+   * then dispose of any parameter blocks that had it selected.
+   * @protected
+   */
+  deleteSelectedParameter() {
+    const {definitionBlock, workspace} = this.findDefinitionBlockAndWorkspace();
+
+    const procedureModel = (
+      definitionBlock as ProcedureBlock
+    ).getProcedureModel();
+    const parameters = procedureModel.getParameters();
+
+    // Find the parameter variable
+    const variableName = this.getText();
+    const variable = workspace.getVariable(variableName);
+
+    if (variable) {
+      // Find the index of the procedure parameter to delete
+      const paramIndexToDelete = parameters.findIndex(
+        parameter => parameter.getName() === variable.name
+      );
+
+      // Delete the parameter from the procedure model
+      if (paramIndexToDelete !== -1) {
+        procedureModel.deleteParameter(paramIndexToDelete);
+      }
+
+      const parameterBlocks = workspace
+        .getAllBlocks()
+        .filter(block => block.type === 'parameters_get');
+      parameterBlocks.forEach(paramBlock => {
+        const varField = paramBlock?.getField('VAR') as FieldVariable | null;
+        if (varField && varField.getVariable()?.name === variable.name) {
+          paramBlock.dispose(true);
+        }
+      });
+    } else {
+      console.error(`Variable with name ${variableName} not found`);
+    }
+  }
+
+  /**
+   * Find the selected variable and rename it, then rename the parameter.
+   * @param {string} newName The new name for the parameter.
+   * @protected
+   */
+  renameSelectedParameter(newName: string) {
+    const {definitionBlock, workspace} = this.findDefinitionBlockAndWorkspace();
+
+    const variableName = this.getText();
+    const variable = workspace.getVariable(variableName);
+    if (variable) {
+      workspace.renameVariableById(variable.getId(), newName);
+
+      const procedureModel = (
+        definitionBlock as ProcedureBlock
+      ).getProcedureModel();
+      const parameters = procedureModel.getParameters();
+
+      // Find the parameter to rename
+      const paramToRename = parameters.find(
+        parameter => parameter.getName() === variableName
+      );
+
+      // Rename the parameter in the procedure model
+      if (paramToRename) {
+        paramToRename.setName(newName);
+      }
+      // In case the block is disconnected from the definition, manually
+      // reset its field value.
+      this.getSourceBlock()?.setFieldValue(variable.getId(), 'VAR');
+    }
+  }
+  /**
+   * For a given parameter block, find the definition block and its workspace.
+   * If the block is part of the function definition, we want the root (top-most) block.
+   * If this block is in a mini-toolbox, we want the flyout's parent block.
+   * If neither of these are true (e.g. the block is disconnected on the workspace,
```

Comment:
super nit: close parentheses after workspace?

Reply:
Fix pushed to the other branch so that I could merge this one without waiting for drone again. https://github.com/code-dot-org/code-dot-org/pull/59201/commits/8bcfea38b7dbfc80340b18625c604256eeab5524